docs/fuel-sync-design.md

```diff
@@ -0,0 +1,345 @@
+# The `fuel-sync` design the version for PoA
+
+## Changelog
+
+- TODO: Date of PR creation/approval
+
+## Glossary
+
+- FS - Fuel sync -> `fuel-sync`
+- P2P - Fuel p2p -> `fuel-p2p`
+- BI - Fuel block importer -> `fuel-block-importer`
+- FC - Fuel core -> `fuel-core`
+- CM - Consensus module
+- Sealed data - The data(blocks, transactions, headers, etc.) signed by the block producer(PoA) or by the consensus comity(PoS)
+- ABGT - Average block generation time in seconds
+- TP - Transactions pool -> `fuel-txpool`
+
+## Overview
+
+The file describes the possible workflow of the FS service 
+and its relationship with P2P and BI.
+During the review process, team must select which way to go, and the document's 
+final version will contain only the selected way with a proper overview.
+
+The first final version of the document will describe the implementation 
+of the FS for the PoA version of the FC. It may contain our assumptions 
+and ideas for the next evolution of the FS for PoS(or maybe FS will 
+have several evolution points).
+
+All names are not final and can be suggested by the reviewers=)
+
+## Relationship with P2P and BI
+
+![img.png](diagrams/fuel-core-design-fuel-sync-part.png)
+
+The FS is a connection point between:
+- In the PoA version P2P and BI
+- In the PoS is also additionally connects P2P with CM
+
+There are three possible ways how to implement those relationships:
+
+1. Every service knows about other services related somehow to it. It 
+can be interpreted as: each service struct containing reference to other 
+services inside and using the `async/await` mechanism to interact.
```

Comment:
Shouldn't we require `async/await` only for operations that are `I/O`? It might be overkill to have everything async, unless ofc everything is I/O

Reply:
Yes, I meant that we could call methods directly. I will rename it to "sync/async".